### PR TITLE
MEB-65: Add Language Selection UI

### PR DESCRIPTION
**Pull request by TheJYKoder**
Implemented language selection dropdown from CritiqueBraniz to MetaBrainz.

### CODE REVIEW

**Comment by paramsingh** [comment]
@brainzbot, add to whitelist

**Comment by paramsingh** [comment]
@TheJYKoder, tests seems to be failing, would you mind taking a look as to why?

**Comment by TheJYKoder** [comment]
Okay, I tried again but I'm unsure if I've updated the manage.py correctly.

File: manage.py [hunk 1]
```diff
@@ -77,7 +77,6 @@ def init_db(force=False, create_db=False):
 
     click.echo("Database has been initialized successfully!")
 
-
```
Comment (anshuman73) [inline manage.py]: Hey, can you add this line back in ?
It's not a big deal, but having two empty lines before a function definition is the preferred formatting according PEP8

File: manage.py [hunk 2]
```diff
@@ -87,8 +86,23 @@ def extract_strings():
     _run_command("pybabel extract -F metabrainz/babel.cfg "
                  "-o metabrainz/messages.pot metabrainz/")
     click.echo("Strings have been successfully extracted into messages.pot file.")
+
```
Comment (anshuman73) [inline manage.py]: Also, maybe add another empty line here, same reason as above :)

**Comment by anshuman73** [comment]
Hi, there's a mild front-end issue - 

screenshot from 2018-01-09 01-12-17

I think a simple fix for this would be to adjust the spacing between the navbar elements. Maybe the padding ?

File: manage.py [hunk 3]
```diff
@@ -77,7 +77,7 @@ def init_db(force=False, create_db=False):
 
     click.echo("Database has been initialized successfully!")
 
-
+
```
Comment (anshuman73) [inline manage.py]: Hey, it seems you added some spaces here by mistake. could you remove them ?

File: metabrainz/static/css/theme/boostrap/list-group.less [hunk 4]
```diff
@@ -21,7 +21,7 @@
 .list-group-item {
   position: relative;
   display: block;
-  padding: 10px 15px;
+  padding: 10px 7.5px;
```
Comment (anshuman73) [inline metabrainz/static/css/theme/boostrap/list-group.less]: Hey, I think editing this file doesn't effect the navbar as it doesn't inherit any styles from this class. Can you revert this back ?

File: metabrainz/static/css/theme/boostrap/variables.less [hunk 5]
```diff
@@ -414,7 +414,7 @@
 //##
 
 //=== Shared nav styles
-@nav-link-padding:                          10px 15px;
+@nav-link-padding:                          10px 7.5px;
```
Comment (anshuman73) [inline metabrainz/static/css/theme/boostrap/variables.less]: editing this file is unnecessary as well, simply the /theme/variables.less is enough as we use that custom file to generate  the .css :)
please revert these back too :)

File: metabrainz/static/css/theme/variables.less [hunk 6]
```diff
@@ -307,7 +307,7 @@
 // Navs
 // -------------------------
 
-@nav-link-padding: 10px 15px;
+@nav-link-padding: 10px 7.5px;
```
Comment (anshuman73) [inline metabrainz/static/css/theme/variables.less]: 7.5 is making it a bit less. It leaves this awkward space in the end.
I feel 10px is just fine, enough space and makes it look decent.

Here's how it looks with 7.5 - 

screenshot from 2018-01-09 23-45-51

And here's with 10px - 

screenshot from 2018-01-09 23-46-33

File: manage.py [hunk 7]
```diff
@@ -87,8 +87,24 @@ def extract_strings():
...
+    languages = ','.join(create_app().config['SUPPORTED_LANGUAGES'])
+    _run_command("tx pull -f -r metabrainz.metabrainz -l %s" % languages)
 
 
+@cli.command()
+def update_strings():
+    """Extract strings and pull translations from Transifex."""
+    extract_strings()
+    pull_translations()
+
```
Comment (anshuman73) [inline manage.py]: Another newline here ;)

**Comment by anshuman73** [comment]
Also, consider squashing your commits into one after making the above changes so that we have a clean history :)

**Comment by anshuman73** [comment]
Can you edit the PR title to conform to out guideliens here -
 https://github.com/metabrainz/guidelines/blob/master/GitHub.md

Just add the ticket name to the title like the way it is suggested in the guidelines. You should be able to find the ticket number in your task description.

**Comment by anshuman73** [comment]
Perfect !
It seems the squashing didn't go through, and rather mixed up some history, but it's fine, we'll squash the commits while merging the PR.
Thanks for your contributions !

**Pull request closed without merging**